dojo/problem/redis.py

```diff
@@ -0,0 +1,143 @@
+        if problem_data:
+            return Problem.from_dict(json.loads(problem_data))
+        return None
+
+    def persist(self, redis_client):
+        if not self.finding_ids:
+            redis_client.hdel("problems", self.problem_id)
+        else:
+            redis_client.hset("problems", self.problem_id, json.dumps(self.to_dict()))
+
+    def update_name_sev(self, finding):
+        if SEVERITY_ORDER[finding.severity] > SEVERITY_ORDER[self.severity]:
+            self.name = finding.title
+            self.severity = finding.severity
+            self.main_finding_id = finding.id
+
+    @staticmethod
+    def add_finding(finding):
+        Problem.remove_finding(int(finding.id))
+        if finding.vuln_id_from_tool and finding.severity != "Info":
+            redis_client = get_redis_client()
+            problem_id = problem_id_b64encode(finding.vuln_id_from_tool)
+            problem = Problem.load_from_id(problem_id, redis_client)
+            if not problem:
+                problem = Problem(problem_id)
+            if finding.id not in problem.finding_ids:
+                problem.update_name_sev(finding)
+                problem.finding_ids.add(finding.id)
+                problem.script_ids.add(finding.vuln_id_from_tool)
+            problem.persist(redis_client)
+            redis_client.hset("id_to_problem", finding.id, problem_id)
+
+    @staticmethod
+    def remove_finding(finding_id):
+        redis_client = get_redis_client()
+        if redis_client.exists("problems") and redis_client.exists("id_to_problem"):
+            problem_id = redis_client.hget("id_to_problem", finding_id)
+            if problem_id:
+                problem = Problem.load_from_id(problem_id, redis_client)
+                if problem:
+                    # This is why we do not use Redis set functionality to store Findings associated with a Problem.
+                    # We need to iterate over Findings anyway whenever a Finding is deleted or changed to update the Problem definition.
+                    problem.finding_ids.remove(finding_id)
+                    if finding_id == problem.main_finding_id:
+                        problem.severity = "Info"
+                        findings = Finding.objects.filter(id__in=problem.finding_ids)
+                        for finding in findings:
+                            problem.update_name_sev(finding)
+
+                    # renew quantity of script_ids if necessary
+                    remaining_script_ids = {finding.vuln_id_from_tool for finding in Finding.objects.filter(id__in=problem.finding_ids)}
+                    problem.script_ids.intersection_update(remaining_script_ids)
+
+                    problem.persist(redis_client)
+                redis_client.hdel("id_to_problem", finding_id)
+
+
+@lru_cache(maxsize=1)
+def get_redis_client():
+    return redis.Redis(host="django-defectdojo-redis-1", port=6379, decode_responses=True)
```

Comment:
Do not hardcode the server's address. I think `redis` might work for all cases, as that is the name of the container. If simply `redis` does not work, create a configuration variable in `settings.py` that people can set to set the Redis server.